kuma/wiki/management/commands/render_document.py

```diff
@@ -2,100 +2,112 @@
 
         if options['all']:
-            logging.info(u"Querying ALL %s documents..." %
-                         Document.objects.count())
-            docs = Document.objects.order_by('-modified').iterator()
-            for doc in docs:
-                self.do_render(doc)
+            # Query all documents, excluding those whose `last_rendered_at` is
+            # within `min_render_age` or NULL.
+            min_render_age = (
+                datetime.datetime.now() -
+                datetime.timedelta(seconds=self.options['min_age']))
+            docs = Document.objects.filter(
+                Q(last_rendered_at__isnull=True) |
+                Q(last_rendered_at__gte=min_render_age))
+            docs = docs.order_by('-modified')
+            docs = docs.values_list('id', flat=True)
+
+            self.chain_render_docs(docs)
 
         else:
             if not len(args) == 1:
-                raise CommandError("Need at least one document path to render")
+                raise CommandError('Need at least one document path to render')
             for path in args:
-                # Accept a single page path from command line, but be liberal in
-                # what we accept, eg: /en-US/docs/CSS (full path); /en-US/CSS (no
-                # /docs); or even en-US/CSS (no leading slash)
+                # Accept a single page path from command line, but be liberal
+                # in what we accept, eg: /en-US/docs/CSS (full path);
+                # /en-US/CSS (no /docs); or even en-US/CSS (no leading slash)
                 if path.startswith('/'):
                     path = path[1:]
                 locale, sep, slug = path.partition('/')
                 head, sep, tail = slug.partition('/')
                 if head == 'docs':
                     slug = tail
-                self.do_render(Document.objects.get(locale=locale, slug=slug))
-
-    def do_render(self, doc):
-        # Skip very recently rendered documents. This should help make it
-        # easier to start and stop an --all command without needing to start
-        # from the top of the list every time.
-        if doc.last_rendered_at:
-            now = datetime.datetime.now()
-            render_age = now - doc.last_rendered_at
-            min_age = datetime.timedelta(seconds=self.options['min_age'])
-            if (render_age < min_age):
-                logging.debug(u"Skipping %s (%s) - rendered %s sec ago" %
-                              (doc, doc.get_absolute_url(), render_age))
-                return
-
-        if self.options['force']:
-            doc.render_started_at = None
+                doc = Document.objects.get(locale=locale, slug=slug)
+                if self.options['force']:
+                    doc.render_started_at = None
+                log.info(u'Rendering %s (%s)' % (doc, doc.get_absolute_url()))
+                try:
+                    render_document(doc, self.cache_control, self.base_url)
```

Comment:
This gave me:
```
...
  File "/data/www/developer.mozilla.org/kuma/kuma/wiki/management/commands/render_document.py", line 90, in handle
    render_document(doc, self.cache_control, self.base_url)
...
  File "/data/www/developer.mozilla.org/kuma/kuma/wiki/tasks.py", line 39, in render_document
    document = Document.objects.get(pk=pk)
...
TypeError: int() argument must be a string or a number, not 'Document'
```
When I ran:
```
./manage.py render_document en-US/docs/User:groovecoder
```
Looks like we need to call it with `doc.pk` like the old `do_render` method did for us?